Neos.Flow/Classes/Core/Booting/Scripts.php

```diff
@@ -736,9 +741,73 @@ public static function buildPhpCommand(array $settings)
             $command .= ' -c ' . escapeshellarg($useIniFile);
         }
 
+        self::ensureWebSubrequestsUseCurrentlyRunningPhpVersion($command);
+
         return $command;
     }
 
+    /**
+     * Compares the realpath of the configured PHP binary (if any) with the one flow was called with in a CLI request.
+     * This avoids config errors where users forget to set Neos.Flow.core.phpBinaryPathAndFilename in CLI.
+     *
+     * @param string phpBinaryPathAndFilename
+     * @throws Exception
+     */
+    protected static function ensureCLISubrequestsUseCurrentlyRunningPhpBinary($phpBinaryPathAndFilename)
+    {
+        // Do nothing for non-CLI requests
+        if (PHP_SAPI !== 'cli') {
+            return;
+        }
+
+        // Resolve any symlinks that the configured php might be pointing to
+        $configuredPhpBinaryPathAndFilename = realpath($phpBinaryPathAndFilename);
+
+        // if the configured PHP binary is empty here, the file does not exist. We ignore that here because it is checked later in the script.
+        if (strlen($configuredPhpBinaryPathAndFilename) === 0) {
```

Comment:
Good catch

Reply:
Oh, and while reading [setup package code](https://github.com/neos/setup/blob/master/Classes/Core/RequestHandler.php#L173-L271), if open_basedir is configured, it may well be that the php binary path is not included, thus regular file functions would not work on that path.

So something like [this](https://github.com/neos/setup/blob/master/Classes/Core/RequestHandler.php#L256-L270) would be required to catch this case.